### ISSUE TITLE
Unchecking Merge Static meshes as skinned causes errors

### ISSUE THREAD

**Issue by Asovrix** [d4rkc0d3r/d4rkAvatarOptimizer#19]
I have quite a complex avatar with some props that I do not want to get merged as meshes, since I suspect it could break stuff.
Unchecking Merge Static Meshes as Skinned or Merge Skinned Meshes causes Merge preview and Debug info to disappear and spews a lot of the same error in console:

> IndexOutOfRangeException: Index was outside the bounds of the array.
> d4rkAvatarOptimizerEditor.OnInspectorGUI () (at Assets/d4rkAvatarOptimizer/src/d4rkAvatarOptimizerEditor.cs:2999)
> UnityEditor.UIElements.InspectorElement+<>c__DisplayClass58_0.<CreateIMGUIInspectorFromEditor>b__0 () (at <a259d3c004024353a2c217da97495055>:0)
> UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

**Edit: after some debugging it seems that even just adding AvatarProp root object from ThatFatKidsMom prefab as an exclusion to optimizer causes above error. Adding "object container" for your prop works fine.**
Link to that prefab: https://github.com/ThatFatKidsMom/Avatar-Prop

**Comment by Asovrix** [comment]
I can still press Create optimized copy, it takes a while because of how large this avatar is, but it produces a copy tagged with `(BrokenCopy)`.

**Comment by d4rkc0d3r** [comment]
Have you tried if it actually does what you want with the settings enabled? When merging static meshes to skinned meshes the vertices of that previously static mesh will get weight painted the transform it was previously on.

**Comment by Asovrix** [comment]
No, it spits out a broken copy. I need to manually exclude props for it to create something not broken, but aside from that I still have unrelated issues... materials turning pink, material losing reference to its maintex only once material is loaded and object that should be toggleable being stuck enabled.
I guess at least for pink material I should create a separate issue?

**Comment by d4rkc0d3r** [comment]
Yeah is def the fact it has 0 materials. I did not account for that to happen.
Thanks for the detailed report, I'll probably fix it tomorrow.

**Comment by d4rkc0d3r** [comment]
As usual get the fix by downloading the current repo form [here](https://github.com/d4rkc0d3r/d4rkAvatarOptimizer/archive/refs/heads/main.zip).
